examples/gobuilder/gobuilder.go

```diff
@@ -0,0 +1,231 @@
+package gobuilder
+
+import (
+	"fmt"
+	"go/build"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/util/system"
+)
+
+func New(src llb.State) *Loader {
+	return newLoader(goBuildBase(), src)
+}
+
+func (l *Loader) Build(pkg string) llb.State {
+	p, err := l.load(pkg)
+	if err != nil {
+		panic(err)
+	}
+	return *p
+}
+
+func goBuildBase() llb.State {
+	goAlpine := llb.Image("docker.io/library/golang:1.8-alpine@sha256:2287e0e274c1d2e9076c1f81d04f1a63c86b73c73603b09caada5da307a8f86d")
```

Comment:
nit: we can use 1.9-alpine@sha256:e91ddbf20f44daeba9a9eca328bc8dbaccf790d26d017dfc572bc003f556e42d